### ISSUE TITLE
Tilemap drawing does not work with certain peering bits.

### ISSUE THREAD

**Issue by m-jocqueviel** [godotengine/godot#82130]
### Godot version

4.1.1

### System information

Godot v4.1.1.stable - Arch Linux #1 SMP PREEMPT_DYNAMIC Wed, 13 Sep 2023 08:37:40 +0000 - Vulkan (Compatibility) - Mesa Intel(R) Graphics (ADL GT2) () - 12th Gen Intel(R) Core(TM) i7-12700H (20 Threads)

### Issue description

While creating a tileset/tilemap, I just couldn't draw certain tiles whose peering bits where "exotic". Here is an example of something I would like to get:

![2023-09-22_18-23](https://github.com/godotengine/godot/assets/18720444/912d83ed-318e-4360-955f-33db5779c043)

So I created a very simple tileset with the most basic tiles plus the ones I would need in my example. When I add the peering bits (by painting the terrain in the tileset) ONLY to the 7 tiles I need, it works fine:

![2023-09-22_18-27](https://github.com/godotengine/godot/assets/18720444/0a87b79c-c087-41af-8247-f58df3a47fed)

But if I try to add the peering bits to the other "standard" tiles, then I just can't get the desired structure (see the new attempt, right to the previous one):

![2023-09-22_18-29](https://github.com/godotengine/godot/assets/18720444/fad25495-9615-4758-a20b-d4af4bcb9b4d)

I checked several times, I'm pretty sure the setup is right, so I guess it is not a desired behaviour?

PS: with different tiles added or removed from the tileset, I get other weird behaviours (tiles constantly changing when REdrawing them, which made my desired structure achievable, but only after redrawing some specific surrounding tiles in a specific order… But I'm not sure it is related, I just wanting to mention it just in case, I might open another issue if it doesn't exist yet.)

### Steps to reproduce

A simple project is provided. Simply open it, open the tilemap scene and reproduce both situations explained in the issue description.

### Minimal reproduction project

[tilemap_issue.zip](https://github.com/godotengine/godot/files/12702805/tilemap_issue.zip)

**Comment by IphStich** [comment]
This could probably be closed. Either as [duplicate](https://github.com/godotengine/godot/issues/89844) (this issue has more discussion) or as [not planned](https://github.com/godotengine/godot/pull/108629#issuecomment-3096904702)

**Comment by groud** [comment]
Yes, I'll close this one as a duplicate. Let me know if https://github.com/godotengine/godot/pull/72030 would help the situation.
